packages/coin-evm/src/utils/arguments.ts

```diff
@@ -51,6 +52,34 @@ async function getSELegacySmartContractTransaction(
   return '15' + path + encoded + chainInfo + chainSignature + formatHex(transaction.data);
 }
 
+async function getSELegacyStakingTransaction(
+  client: LegacyTransaction,
+  chain: ChainProps,
+  coinType: string
+): Promise<string> {
+  const { transaction } = client;
+  const path = await utils.getPath(coinType, client.addressIndex, 5, config.PathType.BIP32);
+  const encoded = Encoder.encodeLegacyTransactionToSE(transaction);
+  const chainInfo = chain.toHexChainInfo();
+  const chainSignature = chain.getSignature();
+
+  const programId = transaction.data.slice(0, 10);
+  let stakingCode = '';
+  if (programId.toLowerCase() === FANTOM.delegateProgram) {
+    stakingCode = '01';
+  } else if (programId.toLowerCase() === FANTOM.withdrawProgram) {
+    stakingCode = '02';
+  } else if (programId.toLowerCase() === FANTOM.undelegateProgram) {
+    stakingCode = '03';
+  } else {
+    throw new error.SDKError(getSELegacyStakingTransaction.name, 'Unrecognized program id for staking contract.');
```

Comment:
What if the other staking methods are required ?

Reply:
Then we update SDK and SCRIPT. The alternative would be allowing all program without recognizing it, i.e. display only [STAKE], which in my opinion a little bit unsafe. Do you have better alternative? I'm open to suggestions.